server/annotation/router.ts

```diff
@@ -0,0 +1,141 @@
+import type { Request, Response } from "express";
+import express from "express";
+import AnnotationCollection from "./collection";
+import * as annotationValidator from "./middleware";
+import * as userValidator from "../user/middleware";
+import * as util from "./util";
+
+const router = express.Router();
+
+/**
+ * Get all the annotations associated with a lecture and author
+ *
+ * @name GET /api/annotation
+ *
+ * @param {string} lectureId - The ID of the lecture to find annotations for
+ * @return {Map<String, Array<UserResponse>>} - Mapping of parent annotation IDs to an array
+ * of [parentAnnotation, ...childAnnotations] sorted by creation date.
+ * @throws {403} - If user is not logged in
+ * @throws {404} - If the lecture ID is not valid
+ *
+ */
+router.get(
+  "/",
+  [userValidator.isUserLoggedIn, annotationValidator.isValidLectureId],
+  async (req: Request, res: Response) => {
+    const annotations = (
+      await AnnotationCollection.findAllByLectureAndAuthor(req.query.lectureId as string, req.session.userId)
+    ).map(util.constructAnnotationResponse);
+    res.status(200).json({
+      message: "Annotations successfully retrieved.",
+      annotations: annotations,
+    });
+  }
+);
+
+/**
+ * Create a new annotation.
+ *
+ * @name POST /api/annotation
+ *
+ * @param {string} lectureId - the ID of the lecture
+ * @param {Number} hour - The hour of the point in video the annotation correponds to
+ * @param {Number} minute - The minute of the point in video the annotation correponds to
+ * @param {Number} second - The second of the point in video the annotation correponds to
+ * @param {string} content - the content of the annotation
+ * @return {AnnotationResponse}
+ * @throws {404} - If lecture or parent ID is not valid
```

Comment:
No parentId here (i see you copied my code)

Reply:
lol yeah got lazy